src/main/scala/org/ergoplatform/nodeView/wallet/Registry.scala

```diff
@@ -0,0 +1,131 @@
+package org.ergoplatform.nodeView.wallet
+
+import io.iohk.iodb.ByteArrayWrapper
+import org.ergoplatform.nodeView.history.ErgoHistory.Height
+
+import scala.collection.mutable
+
+object Registry {
+
+  private val registry = mutable.Map[ByteArrayWrapper, TrackedBox]()
```

Comment:
same as bellow: we need a concurrent data structure here, or all reads should be synchronized as well as writes